test/models/job_execution_test.rb

```diff
@@ -139,8 +141,10 @@
 
   it 'does not export deploy_group information if no deploy groups present' do
     job.update(command: 'env | sort')
-    execute_job
-    job.output.wont_include "DEPLOY_GROUPS"
+    deploy.stub :stage, stage_no_groups do
```

Comment:
I don't think so. The next time a test is run, it will re-run all the `let` statements above, which would run the `Deploy.create!(...)` block and construct a new object.